src/multi.c

```diff
@@ -85,8 +85,14 @@ bool MultiHandleCommand(RedisRaftCtx *rr,
     RaftRedisCommand *cmd = cmds->commands[0];
     size_t cmd_len;
     const char *cmd_str = RedisModule_StringPtrLen(cmd->argv[0], &cmd_len);
+    size_t cmd1_len;
+    const char *cmd1_str = "";
+    if (cmd->argc > 1) { /* TODO: will have to be > 2 in future with magic value in asking for argv[2] */
+        cmd1_str = RedisModule_StringPtrLen(cmd->argv[1], &cmd1_len);
+    }
 
-    if (cmd_len == 5 && !strncasecmp(cmd_str, "MULTI", 5)) {
+    if ((cmd_len == 5 && !strncasecmp(cmd_str, "MULTI", 5)) ||
+       (cmd_len == 6 && !strncasecmp(cmd_str, "ASKING", 6) && cmd1_len == 5 && !strncasecmp(cmd1_str, "MULTI", 5))) {
```

Comment:
```c
    if (cmd_len == 6 && !strncasecmp(cmd_str, "ASKING", 6)) {
        if (cmd->argc > 1) {
            cmd_str = RedisModule_StringPtrLen(cmd->argv[1], &cmd_len);
        }
    }
```

Can't make a suggestion because of the deleted line but would that code above be equivalent to these changes? Checking "asking" just before "multi".

Reply:
somewhat.  it make an "ASKING exec" command valid as well.  I guess thats ok.  the current way is a bit of an artifact of when I was storing an asking flag in the multi struct, which no longer doing, so this simplification is probably ok.  will do that.

Reply:
Yeah, we'll be ignoring any "asking" here with this change. Probably, that's okay.